src/scriptdev2/scripts/kalimdor/durotar.cpp

```diff
@@ -33,89 +33,184 @@ EndContentData */
 
 enum
 {
-    SAY_PEON_AWAKE_1        = -1000795,
-    SAY_PEON_AWAKE_2        = -1000796,
+    SAY_PEON_AWOKEN = -1000795,
 
-    SPELL_PEON_SLEEP        = 17743,
-    SPELL_AWAKEN_PEON       = 19938,
+    SOUND_PEON_WORK = 6197,
+    SOUND_PEON_AWAKE_1 = 6292,
+    SOUND_PEON_AWAKE_2 = 6294,
 
-    NPC_SLEEPING_PEON       = 10556,
-    GO_LUMBERPILE           = 175784,
+    SPELL_PEON_SLEEP = 17743,
+    SPELL_AWAKEN_PEON = 19938,
+
+    NPC_SLEEPING_PEON = 10556,
+    GO_LUMBERPILE = 175784,
 };
 
 struct npc_lazy_peonAI : public ScriptedAI
 {
     npc_lazy_peonAI(Creature* pCreature) : ScriptedAI(pCreature)
     {
         Reset();
-        m_uiStopSleepingTimer = urand(30000, 120000);       // Set on spawn to a potential small timer, to get nice results for initial case
     }
 
-    uint32 m_uiResetSleepTimer;                             // Time, until the npc stops harvesting lumber
-    uint32 m_uiStopSleepingTimer;                           // Time, until the npcs (re)starts working on its own
+    uint32 m_uiSleepTimer; //Time, until the npc goes to sleep
+    uint32 m_uiWakeTimer;  //Time, until the npc wakes up on its own
+    uint32 m_uiChopTimer; //Time, until the npc stops chopping at the tree
+    uint32 m_uiKneelTimer; //Time, delay kneeling to allow sound to finish playing
+    uint32 m_uiGetUpTimer; //Time, npc stays at the pile so kneel animation can finish
+    bool m_bIsAtPile;
+    float m_fSpawnO;
+
+    void RestartWakeTimer() {
```

Comment:
The `{` should have it's own line. I should have shown you this; https://github.com/cmangos/issues/wiki/Coding-Standards